File: playground/steps/parallelMergeState/stepB.step.ts
import { z } from 'zod'
import { EventConfig, StepHandler } from '@motia/core'
import { ParallelMergeStep } from './parallelMerge.types'

type Input = typeof inputSchema

const inputSchema = z.object({})

export const config: EventConfig<Input> = {
  type: 'event',
  name: 'stepB',
  description: 'Hello from Step B',
  subscribes: ['pms.start'],
  emits: ['pms.stepB.done'],
  input: inputSchema,
  flows: ['parallel-merge'],
}

export const handler: StepHandler<typeof config> = async (_, { emit, traceId, state, logger }) => {
  logger.info('[stepB] received pms.start')

  const partialResultB: ParallelMergeStep = { msg: 'Hello from Step B', timestamp: Date.now() }
  await state.set<ParallelMergeStep>(traceId, 'stepB', partialResultB)

  await new Promise((resolve) => setTimeout(resolve, 100))
  await emit({
    type: 'pms.stepB.done',
    data: partialResultB,
  })
}

File: playground/steps/parallelMergeState/stepB.step.py
import time

# Python equivalent of zod schema validation would need to be implemented
# For now using simple dict for config

config = {
    "type": "event",
    "name": "stepB", 
    "description": "Hello from Step B",
    "subscribes": ["pms.start"],
    "emits": ["pms.stepB.done"],
    "flows": ["parallel-merge"]
}

async def handler(_, context):
    context.logger.info("[stepB] received pms.start")

    partial_result_b = {
        "msg": "Hello from Step B",
        "timestamp": int(time.time() * 1000)  # Current timestamp in ms
    }

    context.logger.info("[stepB] setting state")
    await context.state.set(context.trace_id, "stepB", partial_result_b)
    context.logger.info("[stepB] state set")

    await context.emit({
        "topic": "pms.stepB.done",
        "data": partial_result_b
    })

    context.logger.info("[stepB] done")